### PR TITLE
[Improve]Release JDBC connection resources

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: flink-doris-connector/src/main/java/org/apache/doris/flink/connection/SimpleJdbcConnectionProvider.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public SimpleJdbcConnectionProvider(DorisConnectionOptions options) {
 
     @Override
     public Connection getOrEstablishConnection() throws ClassNotFoundException, SQLException {
-        if (connection != null && !connection.isClosed() && connection.isValid(10000)) {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/connection/SimpleJdbcConnectionProvider.java:45]: Why change it to 10?

File: flink-doris-connector/src/main/java/org/apache/doris/flink/lookup/Worker.java [hunk 2]
```diff
@@ -187,8 +187,8 @@ private Map<RecordKey, List<Record>> executeQuery(
             resultRecordMap = new HashMap<>();
             try {
                 long start = System.currentTimeMillis();
-                Connection conn = jdbcConnectionProvider.getOrEstablishConnection();
-                try (PreparedStatement ps = conn.prepareStatement(sql)) {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/lookup/Worker.java:191]: When the upstream tps is high, links may be created frequently?

File: flink-doris-connector/src/main/java/org/apache/doris/flink/catalog/doris/DorisSystem.java [hunk 3]
```diff
@@ -103,10 +104,11 @@ public void createTable(TableSchema schema) {
     }
 
     public void execute(String sql) {
-        try (Statement statement =
-                jdbcConnectionProvider.getOrEstablishConnection().createStatement()) {
+        try (Connection connection = jdbcConnectionProvider.getOrEstablishConnection();
+                Statement statement = connection.createStatement()) {
             statement.execute(sql);
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/catalog/doris/DorisSystem.java:109]: If you close it, you may not be able to reuse the link

File: flink-doris-connector/src/main/java/org/apache/doris/flink/connection/SimpleJdbcConnectionProvider.java (hunk 1 above)
Comment (DongLiang-0) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/connection/SimpleJdbcConnectionProvider.java:45]: Because the unit here is seconds.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**